bundles/jcr/repoinit/src/main/java/org/apache/sling/jcr/repoinit/impl/AclUtil.java

```diff
@@ -54,8 +62,67 @@ public static JackrabbitAccessControlManager getJACM(Session s) throws Unsupport
         return (JackrabbitAccessControlManager) acm;
     }
 
+    /**
+     * checks if provided restriction is supported at
+     * current path or not
+     *
+     * @param specifiedRestrictions
+     * @param jacl
+     * @return
+     */
+    private static void validateRestrictionClauses(List<RestrictionClause> specifiedRestrictions, JackrabbitAccessControlList jacl) throws RepositoryException{
+        String [] supportedRestrictionNames = jacl.getRestrictionNames();
+        Set<String> supportedRestrictionsSet = new HashSet<>(Arrays.asList(supportedRestrictionNames));
+        for(RestrictionClause rc : specifiedRestrictions){
+            if(!supportedRestrictionsSet.contains(rc.getName())) {
+                throw new IllegalStateException("Unsupported restriction "+rc.getName()+" at "+jacl.getPath());
+            }
+        }
+    }
+
+    /**
+     * Converts RestrictionClauses to structure consumable by
+     * jackrabbit
+     * @param list
+     * @param jacl
+     * @param s
+     * @return
+     * @throws RepositoryException
+     */
+    private static LocalRestrictions createOakRestrictions(List<RestrictionClause> list, JackrabbitAccessControlList jacl, Session s) throws RepositoryException {
+        Map<String,Value> restrictions = new HashMap<String,Value>();
+        Map<String,Value[]> mvrestrictions = new HashMap<String,Value[]>();
+
+        if(list != null && !list.isEmpty()){
+            ValueFactory vf = s.getValueFactory();
+
+           for(RestrictionClause rc : list){
+               String restrictionName = rc.getName();
+               int type = jacl.getRestrictionType(restrictionName);
+               Value[] values = new Value[rc.getValues().size()];
+               for(int i=0;i<values.length;i++) {
+                   values[i] = vf.createValue(rc.getValues().get(i),type);
+               }
+
+               if(values.length > 1) {
```

Comment:
@nitin-nizhawan unfortunately that's still missing in the jackrabbit API... but my point wasn't primarily about the case when you have 1 value. it's about an empty value array. as long as there is no way to find out about the mv-status, i would test for value length being 1 (and relying on RestrictionProvider impls to deal with a single value if the target is multivalued).